bucket_map/src/index_entry.rs

```diff
@@ -94,18 +127,18 @@ pub struct IndexEntry<T: Clone + Copy> {
     contents: SingleElementOrMultipleSlots<T>,
 }
 
-/// 62 bits available for ref count
-pub(crate) const MAX_LEGAL_REFCOUNT: RefCount = RefCount::MAX >> 2;
+/// 63 bits available for ref count
+pub(crate) const MAX_LEGAL_REFCOUNT: RefCount = RefCount::MAX >> 1;
 
 /// hold a big `RefCount` while leaving room for extra bits to be used for things like 'Occupied'
 #[bitfield(bits = 64)]
 #[repr(C)]
 #[derive(Debug, Default, Copy, Clone, Eq, PartialEq)]
-struct PackedRefCount {
-    /// tag for Enum
-    slot_count_enum: B2,
-    /// ref_count of this entry. We don't need any where near 62 bits for this value
-    ref_count: B62,
+pub(crate) struct PackedRefCount {
+    /// currently unused
+    pub(crate) unused: B1,
+    /// ref_count of this entry. We don't need any where near 63 bits for this value
+    pub(crate) ref_count: B63,
```

Comment:
it will be used for 'occupied' when ref_count is moved to data buckets. we only need 1 bit for that.
So, we'd end up with 'unused:b1, occupied: b1, ref_count:b62'
I figured we'd just move ref_count to b63.